infra/terraform/global/assets/create_etcd_ebs_snapshot/create_etcd_ebs_snapshot.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/aws"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"os"
+	"log"
+)
+
+var (
+	connection = ec2.New(session.New())
+)
+
+
+func main() {
+
+	instanceKeyFilter   := os.Getenv("INSTANCE_KEY")
+	instanceValueFilter := os.Getenv("INSTANCE_VALUE")
+
+	// Filter instances with role set to master
+	instanceInput := &ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: 	aws.String(fmt.Sprintf("tag:%s", instanceKeyFilter )),
+				Values: []*string{ aws.String(instanceValueFilter)},
+			},
+		},
+	}
+
+	// A reservation corresponds to a command to start instances
+	// A reservation is what you do to provision instances, while an instance is what you get
+	reservation, err := connection.DescribeInstances(instanceInput)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	// Iterate over reservations
+	for index := range reservation.Reservations {
+
+		// Iterate reservations to access ec2 instances
+		for _, inst := range reservation.Reservations[index].Instances {
+
+			// Iterate over each instance's block device metadata
+			for _, blk := range inst.BlockDeviceMappings {
+
+				// Exclude root volumes.  Root volumes have "Delete on termination" set to true mostly
+				rootDevice := *blk.Ebs.DeleteOnTermination
+
+				if rootDevice != true {
+					vol_id := blk.Ebs.VolumeId
+
+					snapshotInput := &ec2.CreateSnapshotInput{
+						VolumeId: vol_id,
+					}
+
+					// Perform ebs snapshot operation
+					snapIt, err := connection.CreateSnapshot(snapshotInput)
+					if err != nil {
+						if aerr, ok := err.(awserr.Error); ok {
+							switch aerr.Code() {
+							default:
+								fmt.Println(aerr.Error())
+							}
+						} else {
+							// Print the error, cast err to awserr.Error to get the Code and
+							// Message from an error.
+							fmt.Println(err.Error())
```

Comment:
It looks to me like both branches of this `if` statement do the same thing, so this could be replaced with
```go
if err != nil {
        fmt.Println(err.Error())
}
```